File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(number)
    results = []
    if number % 3 == 0
      results << "Pling"
    end
    if number % 5 == 0
      results << "Plang"
    end
    if number % 7== 0
      results << "Plong"
    end
    results << number.to_s if results.length == 0
    results.join
  end

  VERSION = 1
end

File: go/src/raindrops/raindrops.go
package raindrops

import (
	"fmt"
	"strings"
)

const testVersion = 2

type intStringPair struct {
	Factor int
	String string
}

var intStringPairs = []intStringPair{
	{3, "Pling"},
	{5, "Plang"},
	{7, "Plong"},
}

func Convert(i int) string {
	return checkFactors(i)
}

func checkFactors(i int) string {
	var result []string
	for _, pair := range intStringPairs {
		if i%pair.Factor == 0 {
			result = append(result, pair.String)
		}
	}
	if len(result) == 0 {
		result = append(result, fmt.Sprintf("%v", i))
	}
	return strings.Join(result, "")
}
